File: typescript-examples/rpa-example/utils/typesAndSchemas.ts
import { z } from "zod";

export const bookConsultationSchema = z.object({
  name: z.string().min(1, "Name is required"),
  email: z.string().email("Valid email is required"),
  phone: z.string().min(1, "Phone is required"),
  date: z.string().min(1, "Date is required"),
  time: z.string().min(1, "Time is required"),
  topic: z.enum(
    [
      "web-scraping",
      "data-extraction",
      "automation",
      "api-integration",
      "other",
    ],
    {
      errorMap: () => ({ message: "Valid topic is required" }),
    }
  ),
});

export const getConsultationsByEmailSchema = z.object({
  email: z.string().email("Valid email is required"),
});

export interface Consultation {
  id: string;
  status: string;
  clientName: string;
  email: string;
  phone: string;
  date: string;
  time: string;
  topic: string;
}

File: python-examples/rpa-auth-example/utils/types_and_schemas.py
from typing import Literal

from pydantic import BaseModel, EmailStr, field_validator


class BookConsultationSchema(BaseModel):
    name: str
    email: EmailStr
    phone: str
    date: str
    time: str
    topic: Literal[
        "web-scraping", "data-extraction", "automation", "api-integration", "other"
    ]

    @field_validator("name", "phone", "date", "time")
    @classmethod
    def validate_not_empty(cls, v: str) -> str:
        if not v or len(v.strip()) < 1:
            raise ValueError("Field is required")
        return v


class GetConsultationsByEmailSchema(BaseModel):
    email: EmailStr


class Consultation(BaseModel):
    id: str
    status: str
    clientName: str
    email: str
    phone: str
    date: str
    time: str
    topic: str
